packages/react-components/src/radio/src/Radio.tsx

```diff
@@ -17,51 +17,68 @@ import {
     useForwardInputApi,
     useSlots
 } from "../../shared";
-import { forwardRef, useImperativeHandle, useMemo, useRef } from "react";
+import { Text } from "../../text";
+import { VisuallyHidden } from "../../visually-hidden";
 import { isNil, isNumber } from "lodash";
 
-const propTypes = {
+export interface InnerRadioProps extends InteractionStatesProps {
     /**
      * A controlled checked state value.
      */
-    checked: bool,
+    checked?: boolean;
     /**
      * The initial value of `checked` when uncontrolled.
      */
-    defaultChecked: bool,
+    defaultChecked?: boolean;
     /**
      * The value to associate with when in a group.
      */
-    value: oneOfType([string, number]).isRequired,
+    value: string | number
```

Comment:
Same as checkbox, as per [MDN](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/radio) the value should only be a string.

(Would also requires a few changes like Checkbox, let me know if you want to take care of it)

Reply:
in other pr